GVRf/Framework/framework/src/main/java/org/gearvrf/GVRAssetLoader.java

```diff
@@ -121,86 +237,144 @@ public void onTextureLoaded(GVRContext context, GVRTexture texture, String texFi
         
         /**
          * Called when a model cannot be loaded.
+         * @param context GVRContext which loaded the texture
          * @param error error message
+         * @param modelFile filename of model loaded
          */
         public void onModelError(GVRContext context, String error, String modelFile)
         {
-            Log.e(TAG, "ERROR: model did not load: %s", error);
+            Log.e(TAG, "ASSET: ERROR: model %s did not load %s", modelFile, error);
             mErrors += "Model " + modelFile + " did not load " + error + "\n";
             mModel = null;
             generateLoadEvent();
         }
         
         /**
          * Called when a texture cannot be loaded.
+         * @param context GVRContext which loaded the texture
          * @param error error message
-         */
+          * @param texFile filename of texture loaded
+        */
         public void onTextureError(GVRContext context, String error, String texFile)
         {
-            Log.e(TAG, "ERROR: cannot load texture %s %s", texFile, error);
+            Log.e(TAG, "ASSET: ERROR: texture did %s not load %s", texFile, error);
             mErrors += "Texture " + texFile + " did not load " + error + "\n";
-            if (mNumTextures > 0) {
+            if (mNumTextures > 0)
+            {
                 --mNumTextures;
-                if (mNumTextures == 0)
+                if ((mNumTextures == 0) && (mModel != null))
                 {
                     generateLoadEvent();
                 }
             }                       
         }
 
+        /**
+         * Called when the model and all of its textures have loaded.
+         * @param context GVRContext which loaded the texture
+         * @param model model that was loaded (will be null if model failed to load)
+         * @param error error messages (will be null if no errors)
+         * @param modelFile filename of model loaded
+         */
         @Override
-        public void onAssetLoaded(GVRContext context, GVRSceneObject model, String filePath, String errors)
+        public void onAssetLoaded(GVRContext context, GVRSceneObject model, String modelFile, String errors)
         {
             mContext.getEventReceiver().removeListener(this);
         }
         
         private void generateLoadEvent()
         {
-            if ((mScene != null) && (mModel != null))
+            if ((mScene != null) && (mModel != null) && (mErrors == ""))
```

Comment:
Please do not compare references to strings; ``"".equals(mErrors)`` would work instead; or if you know mErrors can't be null - ``mErrors.isEmpty()``.